File: test-samples/sample.go
package main

import "fmt"

// Ghost in the VSC - Go テストサンプル

// ========================================
// テスト1: 関数定義
// ========================================
func calculateSum(a, b int) int {
	return a + b
}

// 使い方:
// 1. "calculateSum" をクリック
// 2. 👻が表示される
// 3. 👻にホバー → Lキー
// 4. fmt.Printf("calculateSum: %+v\n", calculateSum) が挿入される

// ========================================
// テスト2: 構造体
// ========================================
type User struct {
	Name string
	Age  int
}

func (u *User) Greet() string {
	return fmt.Sprintf("Hello, %s", u.Name)
}

// ========================================
// テスト3: 変数
// ========================================
func main() {
	result := calculateSum(10, 20)
	user := User{Name: "John", Age: 30}
	
	// "result" をクリック → 👻にホバー → Lキー
	// → fmt.Printf("result: %+v\n", result) が挿入される
}

// ========================================
// テスト4: 選択範囲でのテスト
// ========================================
func processData(data []int) int {
	// 以下の3行を選択してLキーでログ挿入テスト
	filtered := make([]int, 0)
	for _, item := range data {
		if item > 0 {
			filtered = append(filtered, item)
		}
	}
	total := 0
	for _, item := range filtered {
		total += item
	}
	return total
}

// 使い方:
// - 変数や関数をクリック → 👻にホバー → Lキー
// - 構造体の場合は %+v でフィールド名も表示される

File: test-samples/sample.py
# Ghost in the VSC - Python テストサンプル

# ========================================
# テスト1: 関数定義
# ========================================
def calculate_sum(a, b):
    """2つの数値の合計を計算"""
    return a + b

# 使い方:
# 1. "calculate_sum" をクリック
# 2. 👻が表示される
# 3. 👻にホバー → Lキー
# 4. print(f'calculate_sum: {calculate_sum}') が挿入される

# ========================================
# テスト2: 変数
# ========================================
result = calculate_sum(10, 20)
user_name = "John"
is_active = True

# ========================================
# テスト3: クラス
# ========================================
class User:
    def __init__(self, name, age):
        self.name = name
        self.age = age
    
    def greet(self):
        return f"Hello, {self.name}"

# ========================================
# テスト4: 選択範囲でのテスト
# ========================================
def process_data(data):
    # 以下の2行を選択してLキーでログ挿入テスト
    filtered = [item for item in data if item > 0]
    total = sum(filtered)
    return total

# ========================================
# 使用方法
# ========================================
# 変数や関数をクリック → 👻にホバー → キーを押す
# L: print(f'変数名: {変数名}') を挿入
# D: 定義へジャンプ
# R: 参照を表示







